tsconfig.build.json

```diff
@@ -1,7 +1,7 @@
 {
   "compilerOptions": {
     "module": "commonjs",
-    "target": "es2020",
+    "target": "es2022",
```

Comment:
isn't this breaking?

Reply:
We still build for es2020 (check tsup config), it's just the type level support for es2022 (since cause is an es2022 prop)

Alternative is...code checks and casts, which I can do, just let me know

Reply:
ok sounds fine